File: lib/domain/cultivation_plan/interactors/entry_schedule/temperature_requirement_snapshot.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Interactors
      module EntrySchedule
        TemperatureRequirementSnapshot = Struct.new(
          :frost_threshold,
          :optimal_min,
          :optimal_max,
          :base_temperature,
          keyword_init: true
        )
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/interactors/entry_schedule/temperature_requirement_snapshot.rs
//! Ruby: `Domain::CultivationPlan::Interactors::EntrySchedule::TemperatureRequirementSnapshot`

#[derive(Debug, Clone, PartialEq)]
pub struct TemperatureRequirementSnapshot {
    pub frost_threshold: Option<f64>,
    pub optimal_min: Option<f64>,
    pub optimal_max: Option<f64>,
    pub base_temperature: Option<f64>,
}
